File: core/documents/explanation_dependency_engine.py
from __future__ import annotations

from typing import Any, Dict

from core.documents.explanation_chain_engine import build_explanation_chains
from core.evidence import structure_cognition


def reconstruct_explanation_dependencies(text: str) -> Dict[str, Any]:
    chains = build_explanation_chains(text)
    explains = chains.get("reconciled", {}).get("what_explains_what", [])
    observed = {"chains_observed": len(explains)}
    inferred = {"explanation_dependencies": explains}
    reconciled = {"what_explains_what": explains}
    out = structure_cognition(observed, inferred, reconciled, parsed=None)
    out["explanation_dependencies"] = explains
    return out

File: src/documents/explanationDependencyEngine.ts
/**
 * Converted from Python: core/documents/explanation_dependency_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { buildExplanationChains } from "./explanationChainEngine.js";
import { structureCognition } from "../evidence/index.js";

export function reconstructExplanationDependencies(text: any): any {
  var chains: any = buildExplanationChains(text);
  var explains: any = py.get(py.get(chains, "reconciled", {}), "what_explains_what", []);
  var observed: any = {"chains_observed": py.len(explains)};
  var inferred: any = {"explanation_dependencies": explains};
  var reconciled: any = {"what_explains_what": explains};
  var out: any = structureCognition(observed, inferred, reconciled, null);
  py.setItem(out, "explanation_dependencies", explains);
  return out;
}
export { buildExplanationChains, structureCognition };
